storm-core/src/clj/backtype/storm/daemon/worker.clj

```diff
@@ -123,12 +123,13 @@
             port (:port worker)
             storm-cluster-state (:storm-cluster-state worker)
             prev-backpressure-flag @(:backpressure worker)]
-        (if executors 
-          (if (reduce #(or %1 %2) (map #(.get-backpressure-flag %1) executors))
-            (reset! (:backpressure worker) true)   ;; at least one executor has set backpressure
-            (reset! (:backpressure worker) false))) ;; no executor has backpressure set
+        (when executors
+          (reset! (:backpressure worker)
+                  (or @(:transfer-backpressure worker)
```

Comment:
Previously, we have lowWaterMark function as "do nothing" for worker transfer queue's callback, which will not turn off the backpressure when the worker's queue population is below the lowWaterMark. On the other hand, it is possible that all executors all have population above above lowWaterMark but the worker's queue population is still above highWaterMark, and the backpressure flag gets cleared "too often". Anyway, I think this is a good fix, together with the _throttleOn flag in DisruptorQueue.